lib/Configuration.py

```diff
@@ -1100,9 +1100,36 @@ def FindLatestManifest(self, train=None, require_signature=False):
             else:
                 train = temp_mani.Train()
 
+        if enterprise:
+            mani_file = self.TryGetNetworkFile(url="%s/%s/LATEST-ENTERPRISE" % (self.UpdateServerMaster(), train),
+                                               reason="GetLatestManifest")
+            if mani_file is None:
+                log.debug("Could not get latest enterprise manifest file for train %s" % train)
+                return None
+
+            rv = Manifest.Manifest(self, require_signature=require_signature)
+            rv.LoadFile(mani_file)
+            mani_file.close()
+
+            update_rate = rv.dict().get("rate")
+            if update_rate is None:
+                log.debug("Unconditionally accepting an update without a rate")
+                return rv
+
+            if os.path.exists("/data/rate"):
+                with open("/data/rate") as f:
+                    rate = int(f.read())
+            else:
+                rate = int(hashlib.sha1((enterprise_uuid + rv.Version()).encode()).hexdigest(), 16) % 100
+
+            if rate < update_rate:
+                log.debug("Accepting an update with a rate=%d for a system with a rate=%d", update_rate, rate)
+                return rv
+            else:
+                log.debug("Not accepting an update with a rate=%d for a system with a rate=%d", update_rate, rate)
+
```

Comment:
Should this be an else:?  Otherwise mani_file will be set if enterprise is true, and then set again.

Reply:
@kithrup thank you, that makes more sense!